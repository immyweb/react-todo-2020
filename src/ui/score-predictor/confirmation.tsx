import * as React from 'react';

import { colours } from '../../styles/colours';

import Button from '../buttons/buttons';
import { MessageContainer, Heading, Graphic, Copy } from './styles';

interface IProps {
  slideWidth: number;
  onConfirm(): void;
}

export const Confirmation: React.SFC<IProps> = ({ slideWidth, onConfirm }) => {
  return (
    <MessageContainer width={slideWidth}>
      <div>
        <Graphic src="https://www.thesun.co.uk/wp-content/uploads/2018/03/celebrate.png" />
        <Heading fontColour={colours.$football}>NICE ONE!</Heading>
        <Copy>You're ready to go!</Copy>
        <Button background={colours.$football} onClick={onConfirm}>
          See final table
        </Button>
      </div>
    </MessageContainer>
  );
};
