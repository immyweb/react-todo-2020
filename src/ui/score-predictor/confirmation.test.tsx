import * as React from 'react';
import { shallow } from 'enzyme';
import { Confirmation } from './confirmation';
import Button from '../buttons/buttons';

describe('<Confirmation />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Confirmation slideWidth={400} onConfirm={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('onConfirm is called when button is clicked', () => {
    const fn = jest.fn();
    const wrapper = shallow(<Confirmation slideWidth={400} onConfirm={fn} />);
    const btn = wrapper.find(Button);
    btn.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
  });
});
