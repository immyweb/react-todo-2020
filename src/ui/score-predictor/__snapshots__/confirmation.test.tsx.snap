// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Confirmation /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    onConfirm={[Function]}
    slideWidth={400}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <glamorous(img)
          src="https://www.thesun.co.uk/wp-content/uploads/2018/03/celebrate.png"
        />
        <glamorous(glamorous(h3))
          fontColour="#459977"
        >
          NICE ONE!
        </glamorous(glamorous(h3))>
        <glamorous(p)>
          You're ready to go!
        </glamorous(p)>
        <glamorous(button)
          background="#459977"
          onClick={[Function]}
        >
          See final table
        </glamorous(button)>
      </div>,
      "width": 400,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <glamorous(img)
            src="https://www.thesun.co.uk/wp-content/uploads/2018/03/celebrate.png"
          />,
          <glamorous(glamorous(h3))
            fontColour="#459977"
          >
            NICE ONE!
          </glamorous(glamorous(h3))>,
          <glamorous(p)>
            You're ready to go!
          </glamorous(p)>,
          <glamorous(button)
            background="#459977"
            onClick={[Function]}
          >
            See final table
          </glamorous(button)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "src": "https://www.thesun.co.uk/wp-content/uploads/2018/03/celebrate.png",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "NICE ONE!",
            "fontColour": "#459977",
          },
          "ref": null,
          "rendered": "NICE ONE!",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "You're ready to go!",
          },
          "ref": null,
          "rendered": "You're ready to go!",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "background": "#459977",
            "children": "See final table",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "See final table",
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <glamorous(img)
            src="https://www.thesun.co.uk/wp-content/uploads/2018/03/celebrate.png"
          />
          <glamorous(glamorous(h3))
            fontColour="#459977"
          >
            NICE ONE!
          </glamorous(glamorous(h3))>
          <glamorous(p)>
            You're ready to go!
          </glamorous(p)>
          <glamorous(button)
            background="#459977"
            onClick={[Function]}
          >
            See final table
          </glamorous(button)>
        </div>,
        "width": 400,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <glamorous(img)
              src="https://www.thesun.co.uk/wp-content/uploads/2018/03/celebrate.png"
            />,
            <glamorous(glamorous(h3))
              fontColour="#459977"
            >
              NICE ONE!
            </glamorous(glamorous(h3))>,
            <glamorous(p)>
              You're ready to go!
            </glamorous(p)>,
            <glamorous(button)
              background="#459977"
              onClick={[Function]}
            >
              See final table
            </glamorous(button)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "src": "https://www.thesun.co.uk/wp-content/uploads/2018/03/celebrate.png",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "NICE ONE!",
              "fontColour": "#459977",
            },
            "ref": null,
            "rendered": "NICE ONE!",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "You're ready to go!",
            },
            "ref": null,
            "rendered": "You're ready to go!",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "background": "#459977",
              "children": "See final table",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "See final table",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
